services/openai_analysis_service/app.py

```diff
@@ -54,7 +55,7 @@
         log(f"index {summarize_embedding_index} created in pinecone")
     USE_PINECONE = True
 
-summary_config = cfg['analysis_services.openai_analysis_service.summary']
+summary_config = copy.deepcopy(cfg['analysis_services.openai_analysis_service.summary'])
```

Comment:
I think you can just do `cfg.copy([...])` and you get the same thing. But I'm not against an explicit "deep copy" command either.

Reply:
I think it needs to be a deepcopy (copy is shallow).

Reply:
I think dictionary `copy` is deep by default, but I'm going off my memory.

Reply:
Yeah, you are right.

Reply:
Although if you are only popping the top level keys I'm not sure it matters. 
In any case, `deepcopy` is more explicit so I think it is fine.